code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/filter/ServiceCreateValidationFilter.java

```diff
@@ -71,6 +77,14 @@ public Object create(String type, ApiRequest request, ResourceManager next) {
         return super.create(type, request, next);
     }
 
+    protected void validateEnvironment(Service service) {
+        Environment env = objectManager.loadResource(Environment.class, service.getEnvironmentId());
+        if (StringUtils.equals(env.getState(), CommonStatesConstants.ERROR)) {
```

Comment:
@ibuildthecloud said - 

@wlan0 Just add "error" to the CommonStatesConstants and then use that.